File: CS/ViewModels/MainViewModel.cs
using DevExpress.Mvvm;

namespace dxDialog1.ViewModels {
    public class MainViewModel : ViewModelBase {
        protected IDialogService DialogService { get { return this.GetService<IDialogService>(); } }

        public MainViewModel() {
            ShowDialogCommand = new DelegateCommand(ShowDialog);
        }

        public DelegateCommand ShowDialogCommand { get; private set; } 

        public async void ShowDialog() {

            var result = await DialogService.ShowDialogAsync(
                MessageButton.OKCancel,
                "My Dialog Window",
                "DialogView",
                null,
                this);

        }
        
    }
}

File: VB/ViewModels/MainViewModel.vb
Imports DevExpress.Mvvm

Public Class MainViewModel
    Inherits ViewModelBase

    Protected ReadOnly Property DialogService As IDialogService
        Get
            Return Me.GetService(Of IDialogService)()
        End Get
    End Property

    Public Sub New()
        ShowDialogCommand = New DelegateCommand(AddressOf ShowDialog)
    End Sub

    Public Property ShowDialogCommand As DelegateCommand

    Public Async Sub ShowDialog()
        Dim result = Await DialogService.ShowDialogAsync(MessageButton.OKCancel, "My Dialog Window", "DialogView", Nothing, Me)
    End Sub
End Class
